### PR TITLE
Update trilinos compilation instructions

### LINKED ISSUE #2674: Document required Trilinos libraries
When answering [this](https://groups.google.com/forum/#!topic/dealii/rwhHI4Ja-4E) question, I couldn't find anywhere in the documentation that explicitly stated which libraries Trilinos needs to be built with in order to successfully link to it. Judging from [this](https://github.com/dealii/dealii/blob/master/cmake/configure/configure_2_trilinos.cmake) cmake file, I think that the list of prerequisite packages is

- Amesos 
- Epetra 
- Ifpack 
- AztecOO 
- Sacado 
- Teuchos
- ML 
- MueLu (if Trilinos version > 11.14)

Perhaps this should be added to [this](https://www.dealii.org/developer/external-libs/trilinos.html) page?

### PR DESCRIPTION
**Pull request by drwells**
I checked that these flags work with a recent (12.4) version of Trilinos and added some extra notes on MPI and explicit instantiations.

Closes #2674.

### CODE REVIEW

File: doc/external-libs/trilinos.html [hunk 1]
```diff
@@ -55,36 +55,53 @@
...
+    -DCMAKE_INSTALL_PREFIX:PATH=$HOME/share/trilinos \
+    ../
+
+    make install
       </pre>
       You will need to adjust the path into which you want to install Trilinos
-      in the CMAKE_INSTALL_PREFIX line.
+      in the CMAKE_INSTALL_PREFIX line. If you have MPI installed you should
+      also pass the flag <code>-DTPL_ENABLE_MPI=ON</code>
+      to <code>cmake</code>. Additionally, if you computer has enough memory
```
Comment (bangerth) [inline doc/external-libs/trilinos.html]: you -> your

File: doc/external-libs/trilinos.html [hunk 2]
```diff
@@ -55,36 +55,53 @@
...
+    -DBUILD_SHARED_LIBS=ON                           \
+    -DCMAKE_INSTALL_PREFIX:PATH=$HOME/share/trilinos \
+    ../
+
+    make install
       </pre>
       You will need to adjust the path into which you want to install Trilinos
-      in the CMAKE_INSTALL_PREFIX line.
+      in the CMAKE_INSTALL_PREFIX line. If you have MPI installed you should
+      also pass the flag <code>-DTPL_ENABLE_MPI=ON</code>
```
Comment (bangerth) [inline doc/external-libs/trilinos.html]: That's what you already have in the list above. I think we want to encourage people to use MPI, so maybe make the sentence negative: "If you do not want to use MPI (or if your machine simply does not have MPI installed), change the value of the `-DTPL_ENABLE_MPI` to `OFF`."

**Comment by bangerth** [comment]
OK to merge if you address the two minor comments.

File: doc/external-libs/trilinos.html [hunk 3]
```diff
@@ -55,36 +55,53 @@
...
 
-	cd trilinos-12.4.2
-	mkdir build
-	cd build
-
-	cmake \
-	-D Trilinos_ENABLE_Sacado=ON \
-	-D Trilinos_ENABLE_MueLu:BOOL=ON \
-	-D Trilinos_ENABLE_Stratimikos=ON \
-	-D CMAKE_BUILD_TYPE=RELEASE \
```
Comment (Rombur) [inline doc/external-libs/trilinos.html]: You deleted this line. So what is the default value used by Trilinos? RELEASE or DEBUG?

Comment (davydden) [inline doc/external-libs/trilinos.html]: i would keep it, not sure what is the default. Also officially it is called `Release` (https://cmake.org/cmake/help/v3.0/variable/CMAKE_BUILD_TYPE.html)

Comment (Rombur) [inline doc/external-libs/trilinos.html]: I don't think Trilinos follows the standard on that. All their examples use ```DEBUG``` and ```RELEASE```.

Comment (davydden) [inline doc/external-libs/trilinos.html]: i guess you are right, `RELEASE` is used both in Spack and Candi.

**Comment by drwells** [comment]
Thanks for the feedback. I added `-DCMAKE_BUILD_TYPE=RELEASE` (which does appear to be Trilinos' default).

**Pull request merged**